antarest/study/storage/variantstudy/model/command/create_link.py

```diff
@@ -57,11 +53,130 @@ def validate_series(
+            return False
+        simple_match = self.area1 == other.area1 and self.area2 == other.area2
+        if not equal:
+            return simple_match
+        return (
+            simple_match
+            and self.parameters == other.parameters
+            and self.series == other.series
+            and self.direct == other.direct
+            and self.indirect == other.indirect
+        )
+
+    def match_signature(self) -> str:
+        return str(
+            self.command_name.value + MATCH_SIGNATURE_SEPARATOR + self.area1 + MATCH_SIGNATURE_SEPARATOR + self.area2
+        )
+
+    def _create_diff(self, other: "ICommand") -> List["ICommand"]:
+        other = cast(AbstractLinkCommand, other)
+
+        commands: List[ICommand] = []
+        area_from, area_to = sorted([self.area1, self.area2])
+        if self.parameters != other.parameters:
+            properties = LinkInternal.model_validate(other.parameters or {}).model_dump(
+                mode="json", by_alias=True, exclude_none=True, exclude={"area1", "area2"}
+            )
+            commands.append(
+                UpdateConfig(
+                    target=f"input/links/{area_from}/properties/{area_to}",
+                    data=properties,
+                    command_context=self.command_context,
+                    study_version=self.study_version,
+                )
+            )
+        if self.series != other.series:
+            commands.append(
+                ReplaceMatrix(
+                    target=f"@links_series/{area_from}/{area_to}",
+                    matrix=strip_matrix_protocol(other.series),
+                    command_context=self.command_context,
+                    study_version=self.study_version,
+                )
+            )
+        return commands
+
+    def get_inner_matrices(self) -> List[str]:
+        list_matrices = []
+        for attr in MATRIX_ATTRIBUTES:
+            if value := getattr(self, attr, None):
+                assert_this(isinstance(value, str))
+                list_matrices.append(strip_matrix_protocol(value))
+        return list_matrices
+
+    def save_series(self, area_from: str, area_to: str, study_data: FileStudy, version: StudyVersion) -> None:
+        self.series = self.command_context.generator_matrix_constants.get_link(version=version)
+        assert isinstance(self.series, str)
+        if version < STUDY_VERSION_8_2:
+            study_data.tree.save(self.series, ["input", "links", area_from, area_to])
+        else:
+            study_data.tree.save(
+                self.series,
+                ["input", "links", area_from, f"{area_to}_parameters"],
+            )
+
+    def save_direct(self, area_from: str, area_to: str, study_data: FileStudy, version: StudyVersion) -> None:
+        self.direct = self.command_context.generator_matrix_constants.get_link_direct()
+        assert isinstance(self.direct, str)
+        if version >= STUDY_VERSION_8_2:
```

Comment:
Not really the point of this dev but we should raise an Exception when someone gives "direct" or "indirect" with a study prior to v8.2
This could be validated inside a field_validator now that we have the attribute `study_version` inside the commands.
This way you could remove the version check inside the 2 functions
It could also be good to add tests for this inside your matrix tests

Reply:
In fact it could break the compatibility of the R scripts. We should ask them how they call this command before

Reply:
After checking with script team, I removed those conditions and added a model validator